packages/chain-agnostic-permission/src/adapters/caip-permission-adapter-permittedChains.ts

```diff
@@ -146,31 +154,115 @@ export const setPermittedEthChainIds = (
   return updatedCaveatValue;
 };
 
+/*
+ *
+ *
+ * GENERALIZED GETTERS AND SETTERS
+ *
+ *
+ */
+
+/*
+ *
+ * GETTERS
+ *
+ */
+
 /**
- * Filters the scopes object to only include:
- * - Scopes without references (e.g. "wallet:")
- * - CAIP-2 ChainId scopes for the given chainIDs
+ * Gets all scopes from a CAIP-25 caveat value
  *
- * @param scopesObject - The scopes object to filter.
- * @param chainIds - The CAIP-2 chainIDs to filter for.
- * @returns The filtered scopes object.
+ * @param scopesObjects - The scopes objects to get the scopes from.
+ * @returns An array of InternalScopeStrings.
  */
-const filterChainScopesObjectByChainId = (
-  scopesObject: InternalScopesObject,
-  chainIds: CaipChainId[],
-): InternalScopesObject => {
-  const updatedScopesObject: InternalScopesObject = {};
+export function getAllScopesFromScopesObjects(
+  scopesObjects: InternalScopesObject[],
+): InternalScopeString[] {
+  const scopeSet = new Set<InternalScopeString>();
 
-  Object.entries(scopesObject).forEach(([key, scopeObject]) => {
-    // Cast needed because index type is returned as `string` by `Object.entries`
-    const scopeString = key as keyof typeof scopesObject;
-    if (isWalletScope(scopeString) || chainIds.includes(scopeString)) {
-      updatedScopesObject[scopeString] = scopeObject;
+  for (const scopeObject of scopesObjects) {
+    for (const key of Object.keys(scopeObject)) {
+      scopeSet.add(key as InternalScopeString);
     }
-  });
+  }
 
-  return updatedScopesObject;
-};
+  return Array.from(scopeSet);
+}
+
+/**
+ * Gets all scopes (chain IDs) from a CAIP-25 caveat
+ * This extracts all scopes from both required and optional scopes
+ * and returns a unique set.
+ *
+ * @param caip25CaveatValue - The CAIP-25 caveat value to extract scopes from
+ * @returns Array of unique scope strings (chain IDs)
+ */
+export function getAllScopesFromCaip25CaveatValue(
+  caip25CaveatValue: Caip25CaveatValue,
+): CaipChainId[] {
+  return getAllScopesFromScopesObjects([
+    caip25CaveatValue.requiredScopes,
+    caip25CaveatValue.optionalScopes,
+  ]) as CaipChainId[];
+}
+
+/**
+ * Gets all non-wallet namespaces from a CAIP-25 caveat value
+ * This extracts all namespaces from both required and optional scopes
+ * and returns a unique set.
+ *
+ * @param caip25CaveatValue - The CAIP-25 caveat value to extract namespaces from
+ * @returns Array of unique namespace strings
+ */
+export function getAllNonWalletNamespacesFromCaip25CaveatValue(
```

Comment:
I wasn't sure whether we should make this `getAllNameSpacesFromCaip25CaveatValue` and then have an `excludeWalletNamespaces` option param?

Reply:
oof... this is only used in the connect page right now. what if we just filter out wallet in the component?

Reply:
updated here: https://github.com/MetaMask/core/pull/5609/commits/2286b5dc2d672e36c5e343b6fbc5c550e44b371e